frontend/src/components/grantaccess/form/RoleSelect.vue

```diff
@@ -0,0 +1,62 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+import { ErrorMessage, Field } from 'vee-validate';
+import Dropdown from 'primevue/dropdown';
+import type { FamApplicationRole } from 'fam-app-acsctl-api';
+import type { FamRoleDto } from 'fam-admin-mgmt-api/model';
+
+const props = defineProps({
+    roleId: { type: Number, default: 0 },
+    roleOptions: {
+        type: [Array<FamApplicationRole[]>, Array<FamRoleDto[]>],
+        default: [],
+    },
+    label: { type: String, default: 'Assign a role to the user' },
+    fieldId: { type: String, default: 'roleId' },
+});
+
+const emit = defineEmits(['change', 'resetVerifiedForestClients']);
+
+const computedRoleId = computed({
+    get() {
+        return props.roleId;
+    },
+    set(newRoleId: Number) {
+        emit('change', newRoleId);
+    },
+});
+</script>
+
+<template>
+    <div class="form-field">
+        <label>{{ props.label }}</label>
+        <Field
+            :name="props.fieldId"
+            aria-label="Role Select"
+            v-slot="{ field, handleChange, errorMessage }"
+            v-model="computedRoleId"
+        >
+            <Dropdown
+                :options="roleOptions"
+                optionLabel="role_name"
```

Comment:
also we need to adjust this, so the role options in the "GrantAccess" component is still getting from the api, so the data is like below, use `role_name`, and `role_id`. But we should now switch to get role options from the access attribute, the role option data is different, which is using `id` and `name`. So let's make it dynamic for now, and when we work on ticket #1147, we'll switch to use `id` and `name` for both cases.
<img width="582" alt="Screen Shot 2024-02-27 at 2 29 12 PM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/4379da38-ad08-4e37-8c11-dcf25712bc6d">

<img width="903" alt="Screen Shot 2024-02-27 at 2 27 53 PM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/2fe0b14f-a29e-43bf-896d-d59e8085976b">

Reply:
I am updating the `RoleSelect` component to work dynamically with both `FamApplicationRole` and `FamRoleDto` types. After completing task #1147, this dynamic behavior not going to be needed anymore and should be removed.